app/src/main/java/org/mozilla/focus/activity/MainActivity.java

```diff
@@ -915,7 +915,7 @@ public TabView create() {
         public void onBackStackChanged() {
             final BrowserFragment fragment = getBrowserFragment();
             final Fragment top = MainActivity.this.mainMediator.getTopFragment();
-            if (top == null) {
+            if (top == null || UrlInputFragment.FRAGMENT_TAG.equals(top.getTag())) {
```

Comment:
This will make BrowserFragment show when user enter home and click search? Let's also check if the stackSize is only 1.